src/Avalonia.Visuals/Media/FontFamily.cs

```diff
@@ -184,36 +184,25 @@ public override int GetHashCode()
         {
             unchecked
             {
-                var hash = (int)2186146271;
-
-                if (Key != null)
-                {
-                    hash = (hash * 15768619) ^ Key.GetHashCode();
-                }
-                else
-                {
-                    hash = (hash * 15768619) ^ FamilyNames.GetHashCode();
-                }
-
-                if (Key != null)
-                {
-                    hash = (hash * 15768619) ^ Key.GetHashCode();
-                }
-
-                return hash;
+                return ((FamilyNames != null ? FamilyNames.GetHashCode() : 0) * 397) ^ (Key != null ? Key.GetHashCode() : 0);
```

Comment:
I did this change and then I noticed that it used to be like this (hashing both members) and it was changed to only hash `FamilyNames` when `Key` is null. Was this intended, and if yes - why?

Reply:
Key is only valid for custom fonts and family names only for system fonts so combining them doesn't make sense

Reply:
Btw this looks like a bad merge. The double generation of a hash by key is indeed weird.

Reply:
When we have a custom font we don't need to compare the name because `Key.Source` already has the name included

Reply:
How do we know that we are dealing with a custom font? Just because `Key` is set?

Reply:
Yes exactly. Key has no meaning for system fonts.

Reply:
Your changes make sure hash and equals are produced by the same properties so this is the right way to implement it.

Reply:
Simplified `GetHashCode` and added your comment there so another person wandering into this code won't be confused again :)

Reply:
Just noticed your message, so you say that we actually should calculate it from both properties anyway?

Reply:
It's a bit redundant but would be fine. If you implement the equality by comparing key and family names the hash code should be calculate by key and family names as well. They should just match.

Reply:
Will restore previous version then.